### ISSUE TITLE
Archive Collective

### ISSUE THREAD

**Issue by piamancini** [opencollective/opencollective#1317]
## User story 
As a core contributor I want to disable a collective because the community is not interested in fundraising anymore.

## Best solution for this problem
Ideally the core contributor can log in, go to edit collective and click on disable collective.  That would trigger one of two responses 
a) if collective has no transaction history it would delete collective and all dependencies (like we do now with through forrest)
b) if there is a transaction history, it would give the core contributor an alert saying that the collective's transaction history will remain public, but the collective will be disabled.  
- If no remaining budget it would  then remove the tiers and show it as a deactivated collective - with a color change or a notice at the top, etc. 
- If remaining budget it would prompt the core contributor to submit an expense or choose another collective to make a donation.

## MVP
What will be built as an MVP? 
An MVP would be setting a deActivated flag in the database that we could manually set and that would remove the tiers and add a note saying that this collective has been deactivated by their community.

## Metrics 
_What metrics will be used to measure impact?_
Number of collectives deleted 

## Documentation 
_The goal of the documentation should be to make it easier for others to do this_

**Comment by znarf** [comment]
Specification
==========

Frontend side
-------------

Add a form to archive a collective

- This should be an additional form in the "Advanced" section of `EditCollectiveForm`
- This should be using a new separated component: ie: `ArchiveCollective`
- The Component should use the following layout:

```
Archive this Organization|Collective
------------------------------------

Mark this organization|collective as archived, delete all tiers and cancel all subscriptions.

 --------------------------------------
| Archive this organization|collective |
 --------------------------------------
```

- There needs to be a confirmation screen (before the action: are you sure you want to do that?)
- On success, a banner appears on top and the black/white effect is applied to the profile.
- Don't forget to make the component localizable

When browsing a deactivated Collective/Organization:
- Add a top banner explaining the collective is deactivated
- Apply a black/white effect to the profile

API side
--------

- Add a `archivedAt` column to the collective table
- Add a GraphQL mutation to archive a collective.
- Add a property to the GraphQL collective object to know if a collective had been archived.

When archiving a collective, behind the scene (synchronous today / async later):
- check that the authenticated user is an admin
- `archivedAt=now()`, `isActive=false`
- delete all tiers in the collective
- cancel all subscriptions to the collective

**Comment by flickz** [comment]
@znarf  `isActive==false` does not necessarily means the collective has been archived or does it? can we have another column as maybe `archived` that saves a boolean as well?

**Comment by znarf** [comment]
@flickz I was originally thinking about using a new archivedAt column for that.

**Comment by flickz** [comment]
We might be adding a little complexity using date, compare to using a straight forward boolean, leaving `archivedAt` for just information purpose.

**Comment by znarf** [comment]
@flickz It's what we're doing for deletions, and in other places, I personally like it like that.

**Comment by flickz** [comment]
> If no remaining budget it would then remove the tiers and show it as a deactivated collective - with a color change or a notice at the top, etc.
> If remaining budget it would prompt the core contributor to submit an expense or choose another collective to make a donation.

@znarf This details seem to be missing in the specification is that, on purpose? should I handle that myself?

**Comment by znarf** [comment]
@flickz This is too complicated for the first version. We can implement that as follow up enhancements.

**Comment by flickz** [comment]
Closed in [#1782](https://github.com/opencollective/opencollective-api/pull/1782) and [#1520](https://github.com/opencollective/opencollective-frontend/pull/1520).
